From 06d4b76a364d9b3d183112feaaf2f925d72b6aba Mon Sep 17 00:00:00 2001
Message-Id: <RFC-patch-3.4-06d4b76a364-20220711T110019Z-avarab@gmail.com>
In-Reply-To: <RFC-cover-0.4-00000000000-20220711T110019Z-avarab@gmail.com>
References: <YstJl+5BPyR5RWnR@tapette.crustytoothpaste.net>
	<RFC-cover-0.4-00000000000-20220711T110019Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Mon, 11 Jul 2022 11:45:04 +0200
Subject: [RFC PATCH 3/4] .clang-format: do not enforce a ColumnLimit
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Our Documentation/CodingGuidelines mention that "We try to keep to at
most 80 characters per line", but in reality we have a lot of code
that runs afoul of that rule.

Before & after this change running "make style-all-diff-apply" will
yield:

	579 files changed, 32065 insertions(+), 29818 deletions(-)
	509 files changed, 13042 insertions(+), 12745 deletions(-)

As with the preceding change what this leaves us with an unresolved
question though, should we have some stricter version of "make
style-all" that incorporates "ColumnLimit: 80", or perhaps apply it
only on "make style", but then what if someone modifies code that
happens to e.g. search/replace a line running afoul of the limit?

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 .clang-format | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.clang-format b/.clang-format
index 87398d24d4f..5a106d959be 100644
--- a/.clang-format
+++ b/.clang-format
@@ -12,7 +12,7 @@ UseTab: Always
 TabWidth: 8
 IndentWidth: 8
 ContinuationIndentWidth: 8
-ColumnLimit: 80
+ColumnLimit: 0
 
 # C Language specifics
 Language: Cpp
-- 
2.37.0.913.g189dca38629

